gpflow/base.py

```diff
@@ -39,15 +39,17 @@ def __init__(self,
                  *,
                  transform: Optional[Transform] = None,
                  prior: Optional[Prior] = None,
+                 prior_on_x: bool = True,
```

Comment:
Can we make this more obvious? `unconstrained_space_prior=False` maybe? I would expect the default to be that you put priors on the space on which you can easily reason about, i.e. constrained space with "physical" units.